server.js

```diff
@@ -1,38 +1,180 @@
-// to get started!
-// 
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
+import booksData from './data/books.json'
+
+dotenv.config()
 
 const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo"
 mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
 mongoose.Promise = Promise
 
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
-const port = process.env.PORT || 8080
+const bookSchema = new mongoose.Schema({
+  bookID: Number,
+  title: {
+    type: String,
+    lowercase: true
+  },
+  authors: {
+    type: mongoose.Schema.Types.ObjectId,
+    ref: 'Author',
+    lowercase: true
+  },
+  average_rating: Number,
+  isbn13: Number,
+  language_code: {
+    type: String,
+    lowercase: true
+  },
+  num_pages: Number,
+  ratings_count: Number,
+  text_reviews_count: Number
+})
+
+const Book = mongoose.model('Book', bookSchema)
+
+const authorSchema = mongoose.Schema({
+  authors: String
+})
+
+const Author = mongoose.model('Author', authorSchema)
+
+if (process.env.RESET_DB) {
+  console.log('SEEDING!')
+  const seedDB = async () => {
+    await Book.deleteMany()
+    await Author.deleteMany()
+
+    const authorArray = []
+
+    booksData.forEach(async (item) => {
+      const newAuthor = new Author(item)
+      authorArray.push(newAuthor)
+      await newAuthor.save()
+    })
+
+    booksData.forEach(async (item) => {
+      const newBook = new Book({
+        ...item,
+        authors: authorArray.find((singleAuthor) => singleAuthor.authors === item.authors)
+      })
+      await newBook.save()
+    })
+  }
+  seedDB()
+}
+
+const port = process.env.PORT || 8093
 const app = express()
 
-// Add middlewares to enable cors and json body parsing
 app.use(cors())
 app.use(bodyParser.json())
 
-// Start defining your routes here
+app.use((req, res, next) => {
+  if (mongoose.connection.readyState === 1) {
+    next()
+  } else {
+    res.status(503).json({ error: 'Service unavailable' })
+  }
+})
+
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.send(listEndpoints(app))
+})
+
+// Endpoint to get all books and search for title by query param
+app.get('/books', async (req, res) => {
+  const { title } = req.query
+
+  try {
+    if (title) {
+      const books = await Book.find({ 
+        title: {
+          $regex: new RegExp(title, 'i')
+        }
+      }).populate('authors')
+      res.json(books)
+    } else {
+      const books = await Book.find().populate('authors')
+      res.json(books)
+    }
+  } catch (error) {
+    res.status(404).json({ error: 'Something went wrong', details: 'error' })
+  }
+})
+
```

Comment:
Clean and readable code for your routes and endpoints, nice use of queries. ╰(*°▽°*)╯